.github/workflows/release.yml

```diff
@@ -0,0 +1,145 @@
+name: Release
+
+on:
+  # TODO: remove
+  push:
+    branches: [ robert/build/tests ]
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version of this deployment'
+        required: true
+
+env:
+  PYTHON_VERSION: "3.11"
+  POETRY_VERSION: "1.4.2"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+
+      - name: Set Version
+        run: poetry version ${{ github.event.inputs.version }}
+
+      - name: Build package
+        run: poetry build
+
+      - name: Upload build
+        uses: actions/upload-artifact@v4
+        with:
+          name: dist
+          path: dist/
+
+  test-pypi-publish:
+    needs: build
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+
+      - uses: actions/download-artifact@v4
+        with:
+          name: dist
+          path: dist/
+
+      - name: Publish to TestPyPI
+        env:
+          POETRY_PYPI_TOKEN_TESTPYPI: ${{ secrets.TESTPYPI }}
+        run: poetry config repositories.test-pypi https://test.pypi.org/legacy/; poetry config pypi-token.test-pypi $POETRY_PYPI_TOKEN_TESTPYPI; poetry publish --repository test-pypi
```

Comment:
it dawns on me that if I can't get access to the official `redisvl` test-pypi index, we will have to comment out the pre-checks here because right now this leans on the custom `redisvl-test` package which we won't be able to support. Although maybe there is a setting for this in the pyproject.toml

Reply:
yeah we're unfortunately still reliant on Sam unless we can get that oicd work around to happen